### PR TITLE
[DCOS-39710] Investigate kafka acl test failures

### PR DESCRIPTION
**Pull request by mockingbird2**
Extracted the failure detection from #2636 to better track the changes.

### CODE REVIEW

File: frameworks/kafka/tests/auth.py [hunk 1]
```diff
@@ -170,6 +170,10 @@ def write_failed(output) -> bool:
             if "LEADER_NOT_AVAILABLE" in stderr:
                 LOG.warning("Write failed with LEADER_NOT_AVAILABLE")
                 return True
+            if not is_not_authorized(output):
```
Comment (elezar) [inline frameworks/kafka/tests/auth.py]: Note that `output` is a `tuple` and `is_not_authorized` accepts a string. *Yay for interpreted dynamic typed languages*

File: frameworks/kafka/tests/auth.py [hunk 2]
```diff
@@ -170,6 +170,10 @@ def write_failed(output) -> bool:
...
+                LOG.warning("Write failed with AuthorizationException")
+                return False
+            raise Exception("Not yet captured error while sending")
```
Comment (elezar) [inline frameworks/kafka/tests/auth.py]: What about `"Unexpected error while sending"`?

Comment (elezar) [reply frameworks/kafka/tests/auth.py]: Does reading need similar changes?

Comment (mockingbird2) [reply frameworks/kafka/tests/auth.py]: In the read method, all the conditional branches are handled.

**Pull request merged**